docs/lang/articles/kernels/kernel_sync.md

```diff
@@ -0,0 +1,64 @@
+---
+sidebar_position: 2
+---
+
+# Synchronization between Kernels and Python Scope
+
+When using the GPU backend, a kernel is compiled and sent to the GPU queue for execution. The Python program does not wait for the kernel to finish before continuing to execute the next statement. This can be problematic if subsequent computations depend on the result of the kernel. In most of the time, Taichi automatically handles such data dependencies, waiting for the kernel to finish before executing subsequent statements. However, in rare cases, users may need to manually call `ti.sync()` to ensure synchronization.
+
+For example, to measure the execution time of a kernel on the GPU using Python's `time` module:
+
+```python
+import time
+import taichi as ti
+ti.init(arch=ti.gpu)
+
+@ti.kernel
+def benchmark():
+    x = 1.0
+    for i in range(1000):
+        for j in range(10000):
+            x += ti.sin(float(i + j))
+
+start = time.time()
+benchmark()
+end = time.time()
+print(end - start)
+```
+
+The above code defines a `benchmark` function that is a computationally intensive kernel. However, the program does not wait for the kernel to finish before executing subsequent statements, and the execution time of `benchmark` is excluded from the timing measurement.
+
+To wait for the `benchmark` function to finish before executing subsequent statements, we can call `ti.sync()` explicitly. This function blocks the program and waits for all kernel tasks ahead in the GPU queue to finish before executing subsequent statements:
+
+
+```python skip-ci
+import time
+import taichi as ti
+ti.init(arch=ti.gpu)
+
+@ti.kernel
+def benchmark():
+    x = 1.0
+    for i in range(10000):
+        for j in range(100000):
+            x += ti.sin(float(i + j))
+
+start = time.time()
+benchmark()
+ti.sync()
+end = time.time()
+print(end - start)
+```
+
+Most of the time, Taichi automatically handles data synchronization. For example, Taichi will automatically call `ti.sync()` to synchronize data in the following cases:
+
+1. The kernel has a return value.
+2. The kernel contains a `print` statement.
```

Comment:
let's remove this line, cause it's not guaranteed to perform `ti.sync` at the very end of the kernel (Let's say you have a `print` resides somewhere in the middle, so it only syncs everytime it's about to print)